src/Compilers/CSharp/Portable/Compilation/DeconstructionInfo.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    // Deconstructions are represented internally as a tree of conversions, but
+    // since they are not conversions from the language perspective we use a wrapper to
+    // abstract the public API from the implementation.
+
+    /// <summary>
+    /// The representation of a deconstruction as a tree of Deconstruct methods and conversions.
+    /// Methods only appear in non-terminal nodes. All terminal nodes have a Conversion.
+    ///
+    /// Here's an example:
+    /// A deconstruction like `(int x1, (long x2, long x3)) = deconstructable1` with
+    /// `Deconstructable1.Deconstruct(out int y1, out Deconstructable2 y2)` and
+    /// `Deconstructable2.Deconstruct(out int z1, out int z2)` is represented as 5 DeconstructionInfo nodes.
+    ///
+    /// The top-level node has no Method and no Conversion, but has two Nested nodes.
+    /// Its first nested node has no Method, but has a Conversion (Identity).
+    /// Its second nested node has a Method (Deconstructable1.Deconstruct), no Conversion, and two Nested nodes.
```

Comment:
This comment implies that across the summation of all the nodes, there is one node that has a Method, but I would expect two since two methods are being invoked.

Reply:
Argh, sorry about that :-(